hosts/galileo/ssi/__init__.py

```diff
@@ -0,0 +1,334 @@
+                        # unless initialize() is called explicitly.
+
+    # Load the PDS label
+    label = pds3.get_label(filespec)
+
+    # Load the data array
+    vic = vicar.VicarImage.from_file(filespec, extraneous='warn')
+    vicar_dict = vic.as_dict()
+
+    # Get image metadata
+    meta = Metadata(label)
+
+    # Load time-dependent kernels
+    Galileo.load_cks(meta.tstart, meta.tstart + meta.exposure)
+    Galileo.load_spks(meta.tstart, meta.tstart + meta.exposure)
+
+    # Define the field of view
+    FOV = meta.fov(full_fov=full_fov)
+
+    # Trim the image
+    data = meta.trim(vic.data_2d, full_fov=full_fov)
+
+    # Create a Snapshot
+    result = oops.obs.Snapshot(('v','u'), meta.tstart, meta.exposure,
+                               FOV,
+                               path = 'GLL',
+                               frame = 'GLL_SCAN_PLATFORM',
+                               dict = vicar_dict,       # Add the VICAR dict
+                               data = data,             # Add the data array
+                               instrument = 'SSI',
+                               filter = meta.filter,
+                               filespec = filespec,
+                               basename = os.path.basename(filespec))
+
+    result.insert_subfield('spice_kernels',
+                           Galileo.used_kernels(result.time, 'iss',
+                                                return_all_planets))
+
+    return result
+
+#===============================================================================
+def initialize(ck='reconstructed', planets=None, asof=None,
+               spk='reconstructed', gapfill=True,
+               mst_pck=True, irregulars=True):
+    """Initialize key information about the SSI instrument.
+
+    Must be called first. After the first call, later calls to this function
+    are ignored.
+
+    Input:
+        ck,spk      'predicted', 'reconstructed', or 'none', depending on which
+                    kernels are to be used. Defaults are 'reconstructed'. Use
+                    'none' if the kernels are to be managed manually.
+        planets     A list of planets to pass to define_solar_system. None or
+                    0 means all.
+        asof        Only use SPICE kernels that existed before this date; None
+                    to ignore.
+        gapfill     True to include gapfill CKs. False otherwise.
+        mst_pck     True to include MST PCKs, which update the rotation models
+                    for some of the small moons.
+        irregulars  True to include the irregular satellites;
+                    False otherwise.
+    """
+    SSI.initialize(ck=ck, planets=planets, asof=asof,
+                   spk=spk, gapfill=gapfill,
+                   mst_pck=mst_pck, irregulars=irregulars)
+
+
+#===============================================================================
+class Metadata(object):
```

Comment:
Maybe call this GalileoSSIMetadata or something? Metadata seems awfully generic.

Reply:
Couldn't the same be said about from_file()?  If anyone ever wanted access this from outside the Galileo SSI module, they would have to qualify it.  But maybe it should be _Metadata, since it is not meant to be accessed externally.

Reply:
Yes, many of the comments we are discussing need to be addressed as part of the larger issue of cleaning up all of the host modules.  So I'm going to leave this one alone unless you think there's a reason to deal with it now for just this module.